flexmeasures/data/scripts/data_gen.py

```diff
@@ -110,43 +112,92 @@ def add_asset_types(db: SQLAlchemy):
     )
 
 
+def add_weather_sensor_types(db: SQLAlchemy):
+    db.session.add(WeatherSensorType(name="temperature"))
+    db.session.add(WeatherSensorType(name="wind_speed"))
+    db.session.add(WeatherSensorType(name="radiation"))
+
+
+def add_market_types(db: SQLAlchemy):
+    db.session.add(
+        MarketType(
+            name="day_ahead",
+            display_name="day-ahead market",
+            daily_seasonality=True,
+            weekly_seasonality=True,
+            yearly_seasonality=True,
+        )
+    )
+    db.session.add(
+        MarketType(
+            name="tou_tariff",
+            display_name="time-of use tariff",
+            daily_seasonality=True,
+            weekly_seasonality=False,
+            yearly_seasonality=True,
+        )
+    )
+
+
+def add_dummy_tou_market(db: SQLAlchemy):
+    """
+    Add a dummy time-of-use market with a 1-year resolution.
+    Also add a few price points, each covering a whole year.
```

Comment:
Consider noting that we are aware that for this market, the leap years will not have a price on December 31st. To fix that, we need timely-beliefs to switch to defining sensor event resolutions as nominal durations. In practice, we'd use a daily or even hourly resolution until then.

Another workaround may be to set up the event resolution as 366 days, and check how overlapping prices on relevant January 1sts (the ones following a non-leap year) would be shown in the UI and handled by the API.

Reply:
I added a comment for now.

I believe with resolution=366 days, there might be two prices being valid on that one day. This is a situation I can see happening in less exotic circumstances as well. Reality might not always give us data with neat `event_start` points.

I guess a way to handle these situations would be that our querying engine (timely beliefs?) decides that the more recent data point should supersede the other(s).